posts/Yoann/2019-11-08-hibernate-search-quarkus.adoc

```diff
@@ -0,0 +1,1428 @@
+
+[source]
+----
+$ docker build -f src/main/docker/Dockerfile.multistage -t quarkus/hsearch-quarkus .
+[... lots of logs ...]
+Successfully tagged quarkus/hsearch-quarkus:latest
+----
+
+The Docker image is now ready to be used.
+Let's start an environment if it's not already done:
+
+[source]
+----
+$ docker-compose -f environment-stack.yml -p hsearch-quarkus-env up
+----
+
+And once everything is ready, let's start our application:
+
+[source]
+----
+$ docker run --rm -it --network=host \
+        -e POSTGRES_HOST=localhost \
+        -e POSTGRES_DB=hsearch_demo \
+        -e POSTGRES_USER=hsearch_demo \
+        -e POSTGRES_PASSWORD=hsearch_demo \
+        -e ES_HOSTS=http://localhost:9200 \
+        quarkus/hsearch-quarkus
+2019-11-07 10:33:15,520 INFO  [io.quarkus] (main) hsearch-quarkus 1.0-SNAPSHOT (running on Quarkus 1.0.0.CR1) started in 0.849s. Listening on: http://0.0.0.0:8080
+2019-11-07 10:33:15,521 INFO  [io.quarkus] (main) Profile prod activated.
+2019-11-07 10:33:15,521 INFO  [io.quarkus] (main) Installed features: [agroal, cdi, hibernate-orm, hibernate-orm-panache, hibernate-search-elasticsearch, jdbc-postgresql, narayana-jta, resteasy, resteasy-jsonb]
+----
+
+The application started and is ready to accept commands:
+
+[source]
+----
+$ curl -X PUT http://localhost:8080/client/ -H "Content-Type: application/json" -d '{"name":"Wayne Enterprises"}'
+
+{
+    "id": 1,
+    "name": "Wayne Enterprises"
+}
+----
+
+[source]
+----
+$ curl -X GET 'http://localhost:8080/client/search/?terms=enterprise'
+
+[
+    {
+        "id": 1,
+        "name": "Wayne Enterprises"
+    }
+]
+----
+
+== Beyond...
+
+We now are the happy owners of a REST application
+providing both CRUD operations
+and more advanced full-text search operation,
+packaged as a container image.
+
+Because software development is a never-ending task,
+there are still things we could improve:
+
+* improving robustness by taking advantage of https://quarkus.io/guides/flyway[Flyway]
+to handle database schema uppgrade.
+* adding https://quarkus.io/guides/native-and-ssl.html[support for SSL]
+to our native binary if necessary.
+* taking advantage of the many https://quarkus.io/guides/[Quarkus extensions]
+to add a security layer, tracing, fault tolerance, ...
+* adding more search features to our application:
+** we can link:{hsearchDocUrl}#backend-elasticsearch-analysis[tune our analyzer more finely]
+to get better search hits
+(see https://en.wikipedia.org/wiki/Precision_and_recall[precision and recall]).
+** we can link:{hsearchDocUrl}#mapper-orm-directfieldmapping[index more than just text],
+including enums, numbers, date/time values, or even spatial coordinates (points).
+** even custom types can be indexed thanks to link:{hsearchDocUrl}#mapper-orm-bridge[custom bridges]
+** link:{hsearchDocUrl}#search-dsl-predicate[other predicates] are available,
+such as `range` ("between"/"greater than"/...), `within` (points within a circle) and many more.
```

Comment:
I would talk about spatial predicates because it's not entirely obvious we have spatial.